src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/issues/issue54/DumpWithoutSpaceTest.kt

```diff
@@ -0,0 +1,65 @@
+package it.krzeminski.snakeyaml.engine.kmp.issues.issue54
+
+import io.kotest.assertions.fail
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.kotest.matchers.string.shouldContain
+import it.krzeminski.snakeyaml.engine.kmp.api.Dump
+import it.krzeminski.snakeyaml.engine.kmp.api.DumpSettings
+import it.krzeminski.snakeyaml.engine.kmp.api.Load
+import it.krzeminski.snakeyaml.engine.kmp.api.LoadSettings
+
+/**
+ * Issue 54: add a space after anchor (when it is a simple key)
+ */
+class DumpWithoutSpaceTest : FunSpec({
+    fun parse(data: String): Any? {
+        val loadSettings = LoadSettings.builder().setAllowRecursiveKeys(true).build();
+        val load = Load(loadSettings);
+        return load.loadOne(data);
+    }
+
+    test("The document does not have a space after the *1 alias") {
+        try {
+            parse(
+                """|--- &1
+                   |hash:
+                   |  :one: true
+                   |  :two: true
+                   |  *1: true""".trimMargin()
+            )
+            fail("Shouldn't reach here!")
+        } catch (e: Exception) {
+            e.message shouldContain "could not find expected ':'"
+        }
```

Comment:
suggestion: you can replace try-catch using `shouldThrow<Exception> {}` function from kotest

Reply:
Right, here: https://github.com/krzema12/snakeyaml-engine-kmp/pull/309/commits/1faf96c932e272abc1b6d820cf004b322ade77e6